docs/index.md

```diff
@@ -4504,6 +4504,9 @@ You don't *have* to use transformers; you can always create and modify
 statements using statement copying, copy/paste, and editing.
 However, transformers simplify making such changes and the
 results are more likely to be correct.
+Transformers are syntax-aware, so transformers
+automatically add or remove parentheses (and parentheses-like symbols) as pairs,
+add spaces around symbols, and in other ways simplify making changes.
```

Comment:
Transforms are actually not very syntax-aware. They have access to the syntax tree, but each node in this tree is just a text without any type info, or without knowing which nodes are parentheses, which are constants, variables and so on. I can add this info if anyone needs for their custom transforms, but it was not necessary for the default transforms. The default transforms do very basic checks before applying them. For example, the Swap transform checks if the selection consists of 3 or 5 parts. If yes, then it swaps either the first with the third, or the second with the fourth. In the latter case the first and the fifth parts are considered as parentheses no mater what those parts actually are. For example, you may use this statement `wff E. x e. A E. y e. A x =/= y` and click the first `E.` in it. Then you may see that all the available transforms (insert, elide and swap) will produce some nonsense as a result.

Reply:
I understand. I'll remove "syntax-aware" here, I think it gives the wrong idea. I can reword this easily I think.